dig.go

```diff
@@ -692,6 +692,36 @@ func isFieldOptional(f reflect.StructField) (bool, error) {
 	return optional, err
 }
 
+type group struct {
+	name    string
+	flatten bool
```

Comment:
Consider exporting fields and methods of unexported types that are meant for
use by other unexported types. This documents the "public" portion of the
private API.

```suggestion
	Name    string
	Flatten bool
```